src/vs/workbench/contrib/terminalContrib/chatAgentTools/browser/bufferOutputPolling.ts

```diff
@@ -215,3 +240,37 @@ export async function assessOutputForErrors(buffer: string, token: CancellationT
 		return 'Error occurred ' + err;
 	}
 }
+
+/**
+ * Returns true if the last line of the output is a prompt asking for user input (e.g., "Do you want to continue? (y/n)").
+ * This is a heuristic and matches common prompt patterns.
+ */
+/**
+ * Returns the kind of input expected by the last line of the output if it appears to be a user prompt.
+ * For example, returns 'y/n', 'yes/no', 'press key', 'choice', etc., or undefined if not a prompt.
+ */
+export function getExpectedUserInputKind(output: string): string | undefined {
+	if (!output) {
+		return undefined;
+	}
+
+	const patterns: { regex: RegExp; kind: string }[] = [
```

Comment:
This is pwsh's confirm impact one, not sure if there are slight variants but this was from trying to `remove-item folder`.

```
[Y] Yes  [A] Yes to All  [N] No  [L] No to All  [S] Suspend  [?] Help (default is "Y"): 
```